scr/main/java/bittercode/util/DatabaseConfig.java

```diff
@@ -0,0 +1,30 @@
+package com.bittercode.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class DatabaseConfig {
+
+    static Properties prop = new Properties();

```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Static initialization of Properties object and loading of configuration in static block can lead to slow class loading and potential performance issues.
</div>
<div id="fix">
<b>Fix:</b> Implement lazy loading of properties using a singleton pattern to improve startup performance and reduce memory usage when the configuration is not needed.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -7,24 +7,27 @@ import java.util.Properties;
  class DatabaseConfig {
 
 -    static Properties prop = new Properties();
 -    static {
 +    private static Properties prop;
 +    private static DatabaseConfig instance;
 +
 +    private DatabaseConfig() {
 +        prop = new Properties();
 +        loadProperties();
 +    }
 +
 +    public static synchronized DatabaseConfig getInstance() {
 +        if (instance == null) {
 +            instance = new DatabaseConfig();
 +        }
 +        return instance;
 +    }
 +
 +    private void loadProperties() {
          ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
          InputStream input = classLoader.getResourceAsStream("application.properties");
 -
          try {
              prop.load(input);
          } catch (IOException e) {
              e.printStackTrace();
          }
      }
 -
 -    public final static String DRIVER_NAME = prop.getProperty("db.driver");
 -    public final static String DB_HOST = prop.getProperty("db.host");
 -    public final static String DB_PORT = prop.getProperty("db.port");
 -    public final static String DB_NAME = prop.getProperty("db.name");
 -    public final static String DB_USER_NAME = prop.getProperty("db.username");
 -    public final static String DB_PASSWORD = prop.getProperty("db.password");
 -    public final static String CONNECTION_STRING = DB_HOST + ":" + DB_PORT + "/" + DB_NAME;
 +    
 +    public String getProperty(String key) {
 +        return prop.getProperty(key);
 +    }
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged